src/client/funcs/play.js

```diff
@@ -1,40 +1,18 @@
 import Discord from "discord.js";
-const {
-    createAudioResource
-} = require("@discordjs/voice");
 
-module.exports = async function play(interaction, guild, client, url) {
+module.exports = async function play(client, interaction, guild, station) {
     let message = {};
     const radio = client.radio.get(guild.id);
-    const resource = createAudioResource(url);
-    radio.connection.subscribe(radio.audioPlayer);
-    radio.audioPlayer.play(resource);
-    resource.playStream
-        .on("readable", () => {
-            client.funcs.logger('Radio', 'Stream started' + " / " + guild.id + " / " + radio.station.name);
-        })
-        .on("finish", () => {
-            client.funcs.logger('Radio', 'Stream finished' + " / " + guild.id);
-            client.funcs.statisticsUpdate(client, guild, radio);
-            radio.connection?.destroy();
-            radio.audioPlayer?.stop();
-            client.radio.delete(guild.id);
-            return;
-        })
-        .on("error", error => {
-            client.funcs.logger('Radio', 'Stream errored');
-            console.error(error);
-            radio.connection?.destroy();
-            radio.audioPlayer?.stop();
-            client.radio.delete(guild.id);
-            return interaction.reply({
-                content: client.messages.errorPlaying,
-                ephemeral: true
-            });
-        });
+    const audioPlayer = client.streamer.listen(station);
+    radio.connection.subscribe(audioPlayer);
+    client.funcs.logger('Radio', guild.id + " / " + "Play" + " / " + radio.station.name);
 
     message.nowplayingDescription = client.messages.nowplayingDescription.replace("%radio.station.name%", radio.station.name);
-    message.nowplayingDescription = message.nowplayingDescription.replace("%radio.station.owner%", radio.station.owner);
+    if(radio.station.name != radio.station.owner){
```

Comment:
no if needed you can do `radio.station.name != radio.station.owner ? radio.station.owner : ""`

Reply:
Now its updated.